packages/react-next/src/components/Slider/Slider.base.tsx

```diff
@@ -1,408 +1,327 @@
 import * as React from 'react';
-import {
-  KeyCodes,
-  css,
-  getId,
-  getRTL,
-  getRTLSafeKeyCode,
-  warnMutuallyExclusive,
-  initializeComponentRef,
-  Async,
-  on,
-  FocusRects,
-} from '../../Utilities';
-import { ISliderProps, ISlider, ISliderStyleProps, ISliderStyles } from './Slider.types';
+import { KeyCodes, css, getRTL, getRTLSafeKeyCode, warnMutuallyExclusive, on, FocusRects } from '../../Utilities';
+import { ISliderProps, ISliderStyleProps, ISliderStyles } from './Slider.types';
 import { classNamesFunction, getNativeProps, divProperties } from '../../Utilities';
 import { Label } from '../../Label';
-
-export interface ISliderState {
-  value?: number;
-  renderedValue?: number;
-}
+import { useId, useBoolean } from '@uifabric/react-hooks';
 
 const getClassNames = classNamesFunction<ISliderStyleProps, ISliderStyles>();
 const COMPONENT_NAME = 'SliderBase';
 export const ONKEYDOWN_TIMEOUT_DURATION = 1000;
 
-export class SliderBase extends React.Component<ISliderProps, ISliderState> implements ISlider {
-  public static defaultProps: ISliderProps = {
-    step: 1,
-    min: 0,
-    max: 10,
-    showValue: true,
-    disabled: false,
-    vertical: false,
-    buttonProps: {},
-    originFromZero: false,
+const useComponentRef = (props: ISliderProps, thumb: React.RefObject<HTMLSpanElement>, value: number | undefined) => {
+  React.useImperativeHandle(
+    props.componentRef,
+    () => ({
+      get value() {
+        return value;
+      },
+      focus() {
+        if (thumb.current) {
+          thumb.current.focus();
+        }
+      },
+    }),
+    [value],
+  );
+};
+
+export const SliderBase = React.forwardRef((props: ISliderProps, ref: React.Ref<HTMLDivElement>) => {
+  let disposables: (() => void)[] = [];
+  const sliderLine = React.useRef<HTMLDivElement>(null);
+  const thumb = React.useRef<HTMLSpanElement>(null);
+  const [useShowTransitions, { toggle: toggleUseShowTransitions }] = useBoolean(true);
+  const id = useId('Slider');
+  const {
+    step = 1,
+    ariaLabel,
+    className,
+    disabled = false,
+    label,
+    max = 10,
+    min = 0,
+    showValue = true,
+    buttonProps = {},
+    vertical = false,
+    valueFormat,
+    styles,
+    theme,
+    originFromZero = false,
+  } = props;
+  const [value, setValue] = React.useState(
```

Comment:
I believe that this will break the controlled slider scenario. If you use `useControllableValue` hook I think that should resolve this. You can also pass the onChange events into that hook so it will manage those as well.

Reply:
It also returns a callback so you can use that instead of setValue below!